pkg/services/auth/jwt.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+var SecretKey string
+
+const ExpirationTime = time.Minute * 15
+
+// Claims is a custom JWT claims that contains some token information
+type Claims struct {
+	jwt.StandardClaims
+	Provider      string `json:"provider"`
+	ProviderToken string `json:"provider_token"`
+}
+
+// Generate generates and signs a new token
+func Generate(secretKey string, expirationTime time.Duration, providerName, providerToken string) (string, error) {
```

Comment:
I had a chat with @josecordaz f2f and we talked about putting `Generate` and `Verify` in an interface so he can mock it out and call it from the `ApplicationServer` more easily.

Reply:
Why mock it out?

Reply:
That way he doesn't have to mess with the internals of this function.  In my mind, Generate unit tests should focus on the internals.  By mocking it out from `server_test.go` he is able to just focus on testing `application.Server.Authenticate(...)`.  From there he is able to test both paths easily:

1. `Generate` returns a valid token
2. `Generate` returns an error

Without the mock, how does he test the sad path?  He can try to force `Generate` into an error state, but that introduces a cascading effect of failures within the test suites that can get annoying very quickly.  If `Generate` evolves because we sign the JWT with a certificate or encrypt information, then the tests in `server_test.go` will most likely fail and need to be updated as well because they are dependent on the internal implementation of `Genereate`.  

It's a trade off, disconnecting the two means as an engineeer updates `Generate` the engineer can focus on making sure that `jwt_test.go` has full coverage and the tests will clearly show which units have broken.  `server_test.go` is still good because ultimately it only cares with a valid token or error is returned.  End-to-end tests can test the full flow.  It's a trade off but it simplifies the code and makes it easier to test in my opinion.